openeo/util.py

```diff
@@ -281,9 +281,78 @@ def get_temporal_extent(*args,
     elif extent:
         assert start_date is None and end_date is None
         start_date, end_date = extent
+    if start_date and not end_date and isinstance(start_date, str):
+        start_date, end_date = string_to_temporal_extent(start_date)
     return convertor(start_date) if start_date else None, convertor(end_date) if end_date else None
 
 
+def string_to_temporal_extent(start_date: str) -> Tuple[dt.date, dt.date]:
+    """Convert a string that represents a year or a month, into a date range.
+
+    For example:
+    "2021" : means all data for 2021
+    "2022-08": means all data for the month of august in 2022.
+
+    If the day is included in the string then we leave it alone.
+    Note that `get_temporal_extent` already handles having one date string,
+    where the day is present in the string.
```

Comment:
Can we also add proper parameter and return type documentation? So that this is rendered properly in api docs:
https://open-eo.github.io/openeo-python-client/api.html#module-openeo.util